File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& numbers, int target) {
        std::unordered_map<int, int> mp;
        for (int i = 0; i < numbers.size(); ++i) {
            auto it = mp.find(target - numbers[i]);
            if (it != mp.end()) {
                return {it->second + 1, i + 1};
            } else {
                mp.insert({numbers[i], i});
            }
        } 
        return {-1, -1};
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left, right = 0, len(numbers) - 1
        while left < right:
            res = numbers[left] + numbers[right]
            if res == target:
                return [left + 1, right + 1]
            if res < target:
                left += 1
            else:
                right -= 1
        return [-1, -1]